services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -657,14 +691,16 @@ private void resolveResourcePermission(KeycloakAuthorizationRequest request,
                 Objects::nonNull).collect(Collectors.toSet());
 
         if (!requestedScopes.isEmpty() && requestedScopesModel.isEmpty()) {
-            throw new CorsErrorResponseException(request.getCors(), "invalid_scope", "One of the given scopes " + permission.getScopes() + " is invalid", Status.BAD_REQUEST);
+            CorsErrorResponseException invalidScopeException = new CorsErrorResponseException(request.getCors(), "invalid_scope", "One of the given scopes " + permission.getScopes() + " is invalid", Status.BAD_REQUEST);
+            fireErrorEvent(request.getEvent(), Errors.INVALID_SCOPE, invalidScopeException);
```

Comment:
I would just use Errors.INVALID_REQUEST. The details should be enough to understand what happened.

Reply:
agreed